Automattic/wp-calypso
# Focused Launch: add Success view

**Editing Toolkit Release Notes**: Added `redirectTo` function when rendering `<FocusedLaunchModal>`

### Changes proposed in this Pull Request

* Add Success view to Focused Launch flow
* Add `isModalDismissible` and `isModalTitleVisible` keys to the `automattic/launch` data store reducer (plus related actions/selectors). These are used to control the behavior of the Focused Launch modal depending on which view is currently active.
* Add `useFocusedLaunchModal()` hook, to expose modal-related variables and actions from the data-store
* Add `redirectTo` function to the `LaunchContext` in order to abstract URL redirection, since it can happen directly from calypso or from the block editor's iframe.
* Add `onboarding-screen-reader-text()` mixin to `@automattic/onboarding` package — used when hiding the Focused Launch modal title
* Fixed a layout bug that was making the focused launch modal content always scroll vertically

### Testing instructions

* Check out the branch on your machine and run in two parallel terminal windows:
  - `yarn && yarn start`
  - `cd apps/editing-toolkit && yarn dev --sync`
* Add an _unlaunched_ site created with `/start` to your sandbox (for convenience, `SITE_ID`)
* Visit `calypso.localhost:3000/page/SITE_ID/home?flags=create/focused-launch-flow` and click on the "Launch" button. The Focused Launch modal should appear.
* At the bottom of the Summary view, click "Launch your site"
* The Success view should appear. Check that all of the specs listed in #47393 are implemented successfully

Bonus:
* Make sure that when in the Summary, Plan Details, and Domain details views, the modal title is visible and the modal is dismissible (by clicking on the "X" button or pressing the "ESC" key)
* Visit `calypso.localhost:3000/page/SITE_ID/home?flags=create/focused-launch-flow-calypso` and make sure that the modal still launches fine

![Screenshot 2020-11-16 at 17 59 32](https://user-images.githubusercontent.com/1083581/99283611-88e4b600-2835-11eb-83b0-9d2a30ced7f0.png)

Fixes #47393
